### PR TITLE
q1: Clean up queries using new query & escape fns, list destructuring.

### PR DESCRIPTION
**Pull request by bpfoley**
Do several `.inc` files here. These are relatively small, but touch many pages

### CODE REVIEW

File: pinc/dpsession_via_php_sessions.inc [hunk 1]
```diff
@@ -134,34 +138,38 @@ function mysql_session_select($sid)
...
-        $query = "
+        $query = sprintf("
             INSERT INTO sessions
                 (sid, expiration, value)
             VALUES
-                ('$sid', '$expiration', '$value')
-            ";
+                ('%s', '%s', '%s')",
+            DPDatabase::escape($sid),
+            $expiration,
```
Comment (chrismiceli) [inline pinc/dpsession_via_php_sessions.inc:158]: do we trust expiration unconditionally?

File: pinc/user_project_info.inc [hunk 2]
```diff
@@ -19,28 +19,28 @@ various places:
...
 {
-    mysqli_query(DPDatabase::get_connection(), "
+    DPDatabase::query("
         INSERT INTO user_project_info
         SET
             username            = '$username',
             projectid           = '$projectid',
             t_latest_home_visit = $timestamp
         ON DUPLICATE KEY UPDATE
             t_latest_home_visit = $timestamp
```
Comment (chrismiceli) [inline pinc/user_project_info.inc:29]: no placeholders here?

File: pinc/dpsession_via_php_sessions.inc (hunk 1 above)
Comment (cpeel) [reply pinc/dpsession_via_php_sessions.inc:158]: No, per my prior code review this needs `%d` in the `sprintf()`.

File: pinc/dpsession_via_php_sessions.inc [hunk 3]
```diff
@@ -134,34 +138,38 @@ function mysql_session_select($sid)
...
     else
     {
-        $query = "
+        $query = sprintf("
             UPDATE sessions
             SET
-                expiration = '$expiration',
-                value = '$value'
-            WHERE sid = '$sid' AND expiration > ".time()
+                expiration = '%s',
```
Comment (cpeel) [inline pinc/dpsession_via_php_sessions.inc:166]: Here too: `%d`

**Review by cpeel — COMMENTED** [review]
@bpfoley - part of the painstaking work of doing this audit is to ensure we use `sprintf()` / `DPDatabase::escape()` / `validate_projectID()` everywhere. We don't just want to do a straightforward search/replace because the point is ensuring proper SQL validation along the way.

File: pinc/dpsession_via_php_sessions.inc (hunk 1 above)
Comment (cpeel) [reply pinc/dpsession_via_php_sessions.inc:158]: @bpfoley - looks like you resolved this but I don't see it updated, did you miss a push on your branch?

File: pinc/user_project_info.inc (hunk 2 above)
Comment (bpfoley) [reply pinc/user_project_info.inc]: Updated. PTAL.

File: pinc/dpsession_via_php_sessions.inc [hunk 4]
```diff
@@ -134,34 +139,38 @@ function mysql_session_select($sid)
...
     if ( $SessionExists == 0 )
     {
-        $query = "
+        $query = sprintf("
             INSERT INTO sessions
                 (sid, expiration, value)
             VALUES
-                ('$sid', '$expiration', '$value')
-            ";
+                ('%s', '%d', '%s')",
```
Comment (cpeel) [inline pinc/dpsession_via_php_sessions.inc:157]: Don't need the quotes for a number (why does MySQL even allow this!?).

File: pinc/dpsession_via_php_sessions.inc [hunk 5]
```diff
@@ -134,34 +139,38 @@ function mysql_session_select($sid)
...
     else
     {
-        $query = "
+        $query = sprintf("
             UPDATE sessions
             SET
-                expiration = '$expiration',
-                value = '$value'
-            WHERE sid = '$sid' AND expiration > ".time()
+                expiration = '%d',
```
Comment (cpeel) [inline pinc/dpsession_via_php_sessions.inc:167]: Don't need the quotes for the number.

File: pinc/user_project_info.inc [hunk 6]
```diff
@@ -221,11 +240,14 @@ function notify_project_event_subscribers( $project, $event, $extras=array() )
     {
         // Take the number of users subscribed to the event
         // and record that as the project's "interest level".
-        $res4 = mysqli_query(DPDatabase::get_connection(), "
+        $sql4 = sprintf("
             UPDATE projects
-            SET int_level = '$n_subscribers'
-            WHERE projectid = '$projectid'
-        ") or die(DPDatabase::log_error());
+            SET int_level = '%d'
```
Comment (cpeel) [inline pinc/user_project_info.inc:245]: No need for the quotes.

**Review by cpeel — COMMENTED** [review]
You have a few linting errors in `user_project_info.inc`.

Tip: `make lint` in `SETUP` is your friend 😄

File: pinc/user_project_info.inc [hunk 7]
```diff
@@ -19,28 +19,37 @@ various places:
...
-    mysqli_query(DPDatabase::get_connection(), "
+    $sql = sprintf("
         INSERT INTO user_project_info
         SET
-            username            = '$username',
-            projectid           = '$projectid',
-            t_latest_home_visit = $timestamp
+            username            = '%s',
+            projectid           = '%s',
+            t_latest_home_visit = %d,
```
Comment (cpeel) [inline pinc/user_project_info.inc:27]: The comma here makes the SQL invalid (found after testing it in my sandbox).

**Review by cpeel — COMMENTED** [review]
Except for this, everything tests out as best as I can tell.

File: pinc/user_project_info.inc (hunk 7 above)
Comment (bpfoley) [reply pinc/user_project_info.inc]: Sorry :(

**Pull request merged**